src/ErrorHandling/SpectreExceptions.hpp

```diff
@@ -0,0 +1,17 @@
+// Distributed under the MIT License.
```

Comment:
Why is the name `SpectreExceptions`? Isn't it a "spectre" exception because it's written inside this repo?

Reply:
Happy to rename; what would you prefer? Exceptions.hpp?

Reply:
Yep, that's what I had in mind. If we have oodles of exceptions (seems unlikely) we can re-evaluate then :)